services/management/ia/alerts_service.go

```diff
@@ -137,27 +151,45 @@ func (s *AlertsService) ListAlerts(ctx context.Context, req *iav1beta1.ListAlert
 				return nil, errors.Wrapf(err, "failed to convert alert rule")
 			}
 		}
-
 		pass, err := satisfiesFilters(alert, rule.Filters)
 		if err != nil {
 			return nil, err
 		}
 
-		if pass {
-			res = append(res, &iav1beta1.Alert{
-				AlertId:   getAlertID(alert),
-				Summary:   alert.Annotations["summary"],
-				Severity:  managementpb.Severity(common.ParseSeverity(alert.Labels["severity"])),
-				Status:    st,
-				Labels:    alert.Labels,
-				Rule:      rule,
-				CreatedAt: createdAt,
-				UpdatedAt: updatedAt,
-			})
+		if !pass {
+			continue
+		}
+
+		pageTotals.TotalItems++
+		if skip > 0 {
```

Comment:
why not preparing whole result slice and then extract a sub-slice with needed offset? Could simplify logic, so no need in such counters.